js_modules/dagit/packages/core/src/runs/RunsRootNew.tsx

```diff
@@ -0,0 +1,290 @@
+                      ? 'The specified run filters are not valid. Please check the filters and try again.'
+                      : 'An unexpected error occurred. Check the console for details.'
+                  }
+                />
+              </Box>
+            );
+          }}
+        >
+          {({pipelineRunsOrError}) => {
+            if (pipelineRunsOrError.__typename !== 'Runs') {
+              return (
+                <Box padding={{vertical: 64}}>
+                  <NonIdealState
+                    icon="error"
+                    title="Query Error"
+                    description={pipelineRunsOrError.message}
+                  />
+                </Box>
+              );
+            }
+
+            return (
+              <>
+                <StickyTableContainer $top={0}>
+                  <RunTable
+                    runs={pipelineRunsOrError.results.slice(0, PAGE_SIZE)}
+                    onAddTag={onAddTag}
+                    filter={filter}
+                    actionBarComponents={
+                      <Box flex={{direction: 'column', gap: 8}}>
+                        {currentTab !== 'all' ? (
+                          <Box flex={{direction: 'row', gap: 8}}>
+                            {filterTokens
+                              .filter((token) => token.token === 'status')
+                              .map(({token, value}) => (
+                                <Tag key={`${token}:${value}`}>{`${token}:${value}`}</Tag>
+                              ))}
+                          </Box>
+                        ) : null}
+                        <RunsFilterInput
+                          tokens={mutableTokens}
+                          onChange={setFilterTokensWithStatus}
+                          loading={queryResult.loading}
+                          enabledFilters={enabledFilters}
+                        />
+                      </Box>
+                    }
+                  />
+                </StickyTableContainer>
+                {pipelineRunsOrError.results.length > 0 ? (
+                  <div style={{marginTop: '16px'}}>
+                    <CursorHistoryControls {...paginationProps} />
+                  </div>
+                ) : null}
+              </>
+            );
+          }}
+        </Loading>
+      </RunsQueryRefetchContext.Provider>
+    </Page>
+  );
+};
+
+// Imported via React.lazy, which requires a default export.
+// eslint-disable-next-line import/no-default-export
+export default RunsRoot;
+
+const RUNS_ROOT_QUERY = gql`
+  query RunsRootQuery($limit: Int, $cursor: String, $filter: RunsFilter!) {
+    pipelineRunsOrError(limit: $limit, cursor: $cursor, filter: $filter) {
+      ... on Runs {
+        results {
+          id
+          ...RunTableRunFragment
+        }
+      }
+      ... on InvalidPipelineRunsFilterError {
+        message
+      }
+      ...PythonErrorFragment
+    }
+  }
+
+  ${RUN_TABLE_RUN_FRAGMENT}
+  ${PYTHON_ERROR_FRAGMENT}
+`;
```

Comment:
Move the gql docs out into their own files that both old and new component files can import from? I'm not sure what weird things might happen if a query is defined identically in multiple places...

Reply:
I ended up just renaming the queries by appending "New" that way I don't need to shuffle around a ton of imports since the new components are required by the old components and the new components would then need to import the queries from the old components which would cause a circular dependency. So yeah renaming them unique is simplest for now and we'll get rid of the duplication once we merge in the new table/filters